modules/storage/storage-account/main.bicep

```diff
@@ -1,446 +1,723 @@
-@description('Deployment Location')
-param location string
+metadata name = 'Azure Storage Account'
+metadata description = 'This Bicep module creates a Storage Account with zone-redundancy, encryption, virtual network access, and TLS version.'
+metadata owner = 'dciborow'
+
+@description('Deployment Location. It defaults to the location of the resource group.')
+param location string = resourceGroup().location
 
 @description('Prefix of Storage Account Resource Name. This param is ignored when name is provided.')
 param prefix string = 'st'
 
 @description('Name of Storage Account. Must be unique within Azure.')
-param name string = '${prefix}${uniqueString(resourceGroup().id, subscription().id)}'
+@maxLength(24)
+@minLength(3)
+param name string = '${prefix}${uniqueString(resourceGroup().id, location)}'
 
-@description('ID of the subnet where the Storage Account will be deployed, if virtual network access is enabled.')
-param subnetID string = ''
+@description('Tags to be applied to the Storage Account.')
+param tags { *: string } = {}
 
-@description('Toggle to enable or disable virtual network access for the Storage Account.')
```

Comment:
I would like to keep this parameter

Reply:
@description('Storage Account SKU Simple - Toggle to enable or disable zone redundancy when using default sku the Storage Account.')
param isZoneRedundant bool = false

@description('Storage Account SKU.')
@allowed([
  'Premium_LRS' # Local
  'Premium_ZRS' # Zone
  'Standard_GRS' # Global
  'Standard_GZRS' # Global Zone
  'Standard_LRS' # Local
  'Standard_RAGRS' # Read Access Global 
  'Standard_RAGZRS' # Read Access Global - Zonal
  'Standard_ZRS' # Zonal
])
param sku string = isZoneRedundant ? 'Standard_ZRS' : 'Standard_LRS'

Reply:
Parameter `isZoneRedundant` added.